wikidatarefisland/pumps/pump.py

```diff
@@ -18,10 +19,25 @@ def run(self, pipe: AbstractPipe, input_file_path: str, output_file_path: str):
         for line in self.storage.getLines(input_file_path):
             output = pipe.flow(line)
             for single_output in output:
-                # TODO: we're opening and closing this on every single line of output
                 self.storage.append(output_file_path, single_output)
 
 
+class DumpReaderPump(AbstractPump):
+    def __init__(self, storage: Storage, batch_size: int):
+        self.storage = storage
+        self.batch_size = batch_size
+
+    def run(self, pipe: AbstractPipe, input_file_path: str, output_file_path: str):
+        values = []
+        for line in self.storage.get_dump_lines(input_file_path):
+            output = pipe.flow(line)
+            for single_output in output:
+                values.append(json.dumps(single_output, ensure_ascii=False))
+                if len(values) >= self.batch_size:
```

Comment:
yeah, in one hand losing 19 item in 80M is nothing. It doesn't even reach one ppm and I like to fix it but hen it would break the test the depends on this behavior to make sure the batching is happening. Let me see what I can do about it.

Reply:
Fixed it now.

Reply:
Cool! Yeah; I noticed because if you run a small test where the output is < batch_size then nothing is returned at all